server/controllers/changeNonResidentialType/details.ts

```diff
@@ -28,14 +28,15 @@ export default class ChangeNonResidentialTypeDetails extends FormInitialStep {
     const cancelLink = `/view-and-update-locations/${prisonId}/${locationId}`
 
     const fields = { ...locals.fields }
-
-    const convertedCellTypes = location.raw?.convertedCellType || []
+    const convertedCellType = req.form.values.convertedCellType || location.raw?.convertedCellType || ''
+    const otherConvertedCellType = req.form.values.otherConvertedCellType || location.raw?.otherConvertedCellType || ''
 
     fields.convertedCellType.items = fields.convertedCellType.items.map((item: FormWizard.Field) => ({
       ...item,
-      checked: convertedCellTypes.includes(item.value), // Use the new variable
+      checked: item.value === convertedCellType,
```

Comment:
Makes sure the correct item is checked. Was reverting before when selecting the 'other' field.